moveit_plugins/moveit_simple_controller_manager/src/moveit_simple_controller_manager.cpp

```diff
@@ -35,131 +35,164 @@
 
 /* Author: Michael Ferguson, Ioan Sucan, E. Gil Jones */
 
-#include <ros/ros.h>
+#include <rclcpp/rclcpp.hpp>
 #include <moveit_simple_controller_manager/action_based_controller_handle.h>
 #include <moveit_simple_controller_manager/gripper_controller_handle.h>
 #include <moveit_simple_controller_manager/follow_joint_trajectory_controller_handle.h>
-#include <moveit/utils/xmlrpc_casts.h>
+#include <boost/algorithm/string/join.hpp>
 #include <pluginlib/class_list_macros.hpp>
 #include <algorithm>
 #include <map>
 
-using namespace moveit::core;
+static const rclcpp::Logger LOGGER = rclcpp::get_logger("moveit.plugins.moveit_simple_controller_manager");
 
-const std::string LOGNAME("SimpleControllerManager");
+// TOOD(JafarAbdi): Should be in moveit/utils .?
+/*
+ * Check if the map of parameters contains all the input keys
+ */
+bool isSubParameter(const std::map<std::string, rclcpp::Parameter>& parameters, const std::vector<std::string>& keys)
+{
+  std::vector<std::string> missing;
+  for (const std::string& key : keys)
```

Comment:
Did you test this? I think the parameter keys would include the full name (not only the parameter name suffix). In general I like the idea, but in the future we will have to switch to declaring all parameters before accessing them, so they can't be missing in the first place. For now we can keep it like this, though.

Reply:
Yep it's working, [here](https://gist.github.com/JafarAbdi/6aee7645628e0e59ce5aff3ff3e2b0fa) an example